lib/infrastructure/datasources/categories/categories_datasource_impl.dart

```diff
@@ -0,0 +1,31 @@
+import 'package:dio/dio.dart';
+import 'package:gymnastic_center/application/key_value_storage/key_value_storage.dart';
+import 'package:gymnastic_center/domain/datasources/categories/categorie_datasource.dart';
+import 'package:gymnastic_center/domain/entities/categories/category.dart';
+import 'package:gymnastic_center/infrastructure/core/constants/environment.dart';
+import 'package:gymnastic_center/infrastructure/mappers/category_mapper.dart';
+import 'package:gymnastic_center/infrastructure/models/categories/category_response.dart';
+
+class CategoriesDatasourceImpl extends CategoryDatasource {
+  final KeyValueStorageService keyValueStorage;
+  final dio = Dio(BaseOptions(baseUrl: Environment.backendApi));
+  CategoriesDatasourceImpl(KeyValueStorageService keyValueStorageI)
```

Comment:
En esta parte del codigo haz unos leves cambios.

1. En el baseUrl pon: `baseUrl: '${Environment.backendApi}/category' ` , de esta forma no colocas eso a cada petición
2. En el constructor pon this.keyValueStorage es mas practico y mas gustoso al leer